### ISSUE TITLE
Every Door errors when using IPv6

### ISSUE THREAD

**Issue by pandadeepimpact** [celzero/rethink-app#1819]
**Device**: Redmi 10C

**OS**: MIUI 13 (Android 11)

**Rethink version**: v0.5.5n (GitHub version)

Whenever I try to log in (pic 1) or download (pic 2) in the mapping app **Every Door**, these errors appear:

<details><summary>screenshots</summary>
img0:

![image](https://github.com/user-attachments/assets/5e5c665c-60de-4798-8ad4-ae6a9550b6b1)

img1:
![{527E053F-A4DA-4722-8E29-9DC7209DEF97}](https://github.com/user-attachments/assets/4ef2f61f-ef45-4fa9-b6a6-62c8db131c2c)
</details>

These happen even with Bypass DNS & Firewall but is resolved by setting the app on Exclude, so it is not a filter issue.

Any idea what these errors mean and how to resolve them?

Please let me know any additional details you need, I'll be glad to help in resolving this issue.

**Comment by ignoramous** [comment]
> Any idea what these errors mean

These errors are due to the app not being able to connect to `api.openstreetmap.org` over HTTPS.

> any additional details you need

Can you search for `api.openstreetmap.org` in both *Configure* -> *Logs* -> swipe to "Network" tab & then the "DNS" tab, and see what comes up? If you tap on the entires in the "Network" & "DNS" tabs, a *bottomsheet* would come up, the *footer* of which should contain any error msgs from Rethink for that connection.

**Comment by Lanius-collaris** [comment]
Was you using an IPv4-only or IPv6-only network? Can you set _IP version_ to auto and enable _Perform connectivity checks_ in _Configure_ -> _Network_?

**Comment by pandadeepimpact** [comment]
> Were you using an IPv4-only or IPv6-only network? Can you set _IP version_ to auto and enable _Perform connectivity checks_ in _Configure_ -> _Network_?

IPv4. Tried doing your suggestion but the same error appears.

Though I noticed some patterns in the logs. That it is the IPv4 addresses, like 172.67.173.161 and 104.21.88.66, that are ok (no error) in the firewall, while the IPv6 addresses, like 2606:4700:3030::ac43:ada1 and 2606:4700:3034::6815:5842, trigger the error (network is unreachable).

Perhaps Every Door alternates between IPv4 and IPv6 for some reason, and whenever it chooses the IPv6 connections, it would trigger errors on Rethink, causing the download to fail? Yet another mapping app, SCEE, uses the same IP addresses and triggers the same errors on Rethink (also network is unreachable on IPv6 addresses), but downloads there never fail.

![Image](https://github.com/user-attachments/assets/d06a7ab3-ffcf-4b2e-815a-811f99c623cd)

**Comment by pandadeepimpact** [comment]
> Strange. Can you confirm if Perform Connectivity checks was turned ON?

> Can you also confirm if Configure -> DNS -> Advanced DNS filtering is turned OFF?

Perform Connectivity checks is ON, but so is Advanced DNS filtering. I'll test turning off Advanced DNS filtering and get back to you.

> To workaround the issue you're seeing wrt IPv6, switch to IPv4 in Configure -> Network -> Choose IP version.

Prior to switching Choose IP version to Auto, I'm already using IPv4. Though I can test whether the combination of IPv4 and turning off Advanced DNS filtering will resolve the issue.

Btw it seems that not only the app Every Door is affected by this issue. I discovered that another app, Seal, suffer from the same problem (downloads failing with error 104).

![Image](https://github.com/user-attachments/assets/b4b6cc77-f005-417c-8b59-660ef450a0ac)

![Image](https://github.com/user-attachments/assets/04ac85f2-ef04-468b-b23b-6fdcca37c169)

With exactly the same symptoms: IPv6 addresses result in "network is unreachable" in firewall, but "no error" in DNS, and using Exclude also resolves the problem.

**Comment by pandadeepimpact** [comment]
You're right @ignoramous, switching Advanced DNS Filtering to OFF resolved the issue. There are also no more errors in both apps.

Though how it was resolved is even stranger, where after turning off Advanced DNS Filtering and restarting my phone, the IPv6 addresses all disappeared in both Every Door and Seal:

Network Logs of Every Door

![Image](https://github.com/user-attachments/assets/ac99fa79-286d-451c-8ace-4745ae13437d)

DNS Logs of Every Door

![Image](https://github.com/user-attachments/assets/74c2aaa7-6043-49ee-bfa0-1269f61f54a7)

Network Logs of Seal

![Image](https://github.com/user-attachments/assets/125b1ad6-fbcb-44b6-8d8b-4d6c380734ed)

DNS Logs of Seal

![Image](https://github.com/user-attachments/assets/66d8d733-e63e-40c3-abe0-be69794cbeea)

Every Door used to have some IPv4 addresses, but Seal used to have only IPv6 addresses. So Advanced DNS Filtering is converting some IPv4 addresses into IPv6 addresses?

**Comment by ignoramous** [comment]
> You're right @ignoramous, switching Advanced DNS Filtering to OFF resolved the issue. There are also no more errors in both apps ... Though how it was resolved is even stranger ...

Nothing strange about it. *Advanced DNS filtering* is a big hairy beast (:

> So Advanced DNS Filtering is converting some IPv4 addresses into IPv6 addresses?

Not really, but kind of. As in, Rethink may engage in 4to6 translations if *Advanced DNS filtering* is turned ON (despite "Choose IP version" set to "IPv4").

We've since "fixed" or now restrict *Advanced DNS filtering* portions in Rethink from doing 4to6 (unless absolutely necessary). You *should* be able to use the improved *Advanced DNS filtering* in `v055o` with Seal & Every Door.

Dups (closing this issue; feel free to re-open):
- https://github.com/celzero/rethink-app/issues/1550
- https://github.com/celzero/rethink-app/issues/1546
- https://github.com/celzero/rethink-app/issues/1115
